lsp-mode.el

```diff
@@ -5322,6 +5328,42 @@ argument ask the user to select which language server to start. "
                (apply 'concat (--map (format "[%s]" (lsp--workspace-print it))
                                      lsp--buffer-workspaces)))))
 
+(defvar lsp--emacs-started nil
+  "A boolean value indicating whether Emacs has finished starting.")
+
+(add-hook 'emacs-startup-hook #'lsp--emacs-startup)
+
+(defun lsp--emacs-startup ()
+  "Run `lsp' for deferred buffers that are visible."
+  (setq lsp--emacs-started t)
+  ;; Now that Emacs is done starting, check if there are visible buffers
+  ;; that were deferred during startup.
+  (dolist (buf (buffer-list))
+    (with-current-buffer buf
+      (when lsp--buffer-deferred (lsp--if-visible)))))
+
+(defun lsp--if-visible ()
```

Comment:
can you rename it to something like lsp--init-if-visible

Reply:
Done.